### ISSUE TITLE
Improve the quality of thinking by introducing entropy reward.

### ISSUE THREAD

**Issue by DeadLining** [om-ai-lab/VLM-R1#270]
I am currently researching whether the introduction of entropy reward can enhance the quality of thinking.

The reward configuration in my study comprises:
- accuracy reward
- format reward
- completion length reward
- entropy reward

I conducted a comparison of completion length dynamics over training steps between models trained with and without entropy reward.

w/o entropy reward
<img width="300" alt="Image" src="https://github.com/user-attachments/assets/b61cb7cb-33e5-4d2b-8626-58519f4dae7c" />

w entropy reward
<img width="300" alt="Image" src="https://github.com/user-attachments/assets/dbf22b31-8655-4e5c-85d0-03ed697d1d86" />

**Comment by lijingcheng2021** [comment]
Can you explain how the entropy reward is implemented, or take a look at the code?"

**Comment by DeadLining** [comment]
> Can you explain how the entropy reward is implemented, or take a look at the code?"

Sure! I added a relative entropy reward to encourage diversity across generated outputs. It works by computing the KL divergence between the token distributions of different completions, then normalizes the scores using softmax. This helps promote variation and reduces output collapse.

**Comment by lijingcheng2021** [comment]
> > Can you explain how the entropy reward is implemented, or take a look at the code?"你能解释一下熵奖励是如何实现的吗，或者看看代码？
> 
> Sure! I added a relative entropy reward to encourage diversity across generated outputs. It works by computing the KL divergence between the token distributions of different completions, then normalizes the scores using softmax. This helps promote variation and reduces output collapse.当然！我添加了相对熵奖励以鼓励生成输出的多样性。它通过计算不同完成结果的词元分布之间的 KL 散度来实现，然后使用 softmax 对分数进行归一化。这有助于促进变化并减少输出崩溃。

Could this lead to a problem where the model's output becomes longer and longer, and could that be harmful?
